File: iSeconds.Droid/Camcorder/CamcorderPreview.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Views;
using Android.Widget;
using iSeconds.Domain;

namespace iSeconds.Droid
{
   [Activity (Label = "CamcorderPreview", Theme = "@android:style/Theme.NoTitleBar.Fullscreen")]         
	public class CamcorderPreview : Activity//ISecondsActivity
   {
		public const Result RetakeResult = Result.FirstUser + 1; 

		protected override void OnCreate (Bundle bundle)
		{
			base.OnCreate (bundle);

			string videoPath = this.Intent.Extras.GetString ("video.path");

			SetContentView(Resource.Layout.CamcorderPreview);
			var videoView = FindViewById<VideoView> (Resource.Id.camcorderPreviewVideo);
			videoView.SetVideoPath (videoPath);
			videoView.Start ();

			Button confirmButton = (Button)this.FindViewById (Resource.Id.camcorderPreviewConfirmButton);
			confirmButton.Click += (object sender, EventArgs e) => {

				this.SetResult(Result.Ok);
				this.Finish();

			};

			Button retakeButton = (Button)this.FindViewById (Resource.Id.camcorderPreviewRetakeButton);
			retakeButton.Click += (object sender, EventArgs e) => {

				this.SetResult(RetakeResult);
				this.Finish();
			};

			Button cancelButton = (Button)this.FindViewById (Resource.Id.camcorderPreviewCancelButton);
			cancelButton.Click += (object sender, EventArgs e) => {

				this.SetResult(Result.Canceled);
				this.Finish();
			};

		}
	}
}

File: jYester/src/iSeconds/Droid/CamcorderPreview.java
package iSeconds.Droid;

import android.app.Activity;
import android.os.Bundle;
import android.view.View;
import android.view.View.OnClickListener;
import android.widget.Button;
import android.widget.VideoView;

public class CamcorderPreview extends Activity {
	
	public static int PreviewResult = 1;
	
	public static final int RetakeResult = Activity.RESULT_FIRST_USER + 1; 

	@Override
	protected void onCreate (Bundle bundle)
	{
		super.onCreate (bundle);

		String videoPath = this.getIntent().getExtras().getString ("video.path");

		setContentView(R.layout.fragment_camcorder_preview);
		VideoView videoView = (VideoView) findViewById (R.id.camcorderPreviewVideo);
		videoView.setVideoPath (videoPath);
		videoView.start ();

		Button confirmButton = (Button)this.findViewById (R.id.camcorderPreviewConfirmButton);
		
		confirmButton.setOnClickListener(new OnClickListener() {
			
			@Override
			public void onClick(View v) {
				CamcorderPreview.this.setResult(Activity.RESULT_OK);
				CamcorderPreview.this.finish();
				
			}
		});
		
		Button retakeButton = (Button)this.findViewById (R.id.camcorderPreviewRetakeButton);
		retakeButton.setOnClickListener(new OnClickListener() {
			
			@Override
			public void onClick(View v) {
				CamcorderPreview.this.setResult(RetakeResult);
				CamcorderPreview.this.finish();

			}
		});

		Button cancelButton = (Button)this.findViewById (R.id.camcorderPreviewCancelButton);
		cancelButton.setOnClickListener(new OnClickListener() {
			
			@Override
			public void onClick(View v) {
				CamcorderPreview.this.setResult(Activity.RESULT_CANCELED);
				CamcorderPreview.this.finish();
				
			}
		});

	}

}
